nomad/structs/config/ui.go

```diff
@@ -21,6 +29,80 @@ type UIConfig struct {
 	Label *LabelUIConfig `hcl:"label"`
 }
 
+// only covers the elements of
+// https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP we need or care about
+type ContentSecurityPolicy struct {
+	ConnectSrc     []string `hcl:"connect_src"`
+	DefaultSrc     []string `hcl:"default_src"`
+	FormAction     []string `hcl:"form_action"`
+	FrameAncestors []string `hcl:"frame_ancestors"`
+	ImgSrc         []string `hcl:"img_src"`
+	ScriptSrc      []string `hcl:"script_src"`
+	StyleSrc       []string `hcl:"style_src"`
+}
+
+// Copy returns a copy of this Vault UI config.
+func (old *ContentSecurityPolicy) Copy() *ContentSecurityPolicy {
+	if old == nil {
+		return nil
+	}
+
+	nc := new(ContentSecurityPolicy)
+	*nc = *old
```

Comment:
In Go this only copies primitives and won't copy `[]string`. So you'll need to do this:

```suggestion
	*nc = *old
	nc.ConnectSrc = slices.Clone(old.ConnectSrc)
	nc.DefaultSrc = slices.Clone(old.DefaultSrc)
	nc.FormAction = slices.Clone(old.FormAction)
	nc.FrameAncestors = slices.Clone(old.FrameAncestors)
	nc.ImgSrc = slices.Clone(old.ImgSrc)
	nc.ScriptSrc = slices.Clone(old.ScriptSrc)
	nc.StyleSrc = slices.Clone(old.StyleSrc)	
```